From: Ivan Vecera <ivecera@redhat.com>
Date: Tue, 15 Feb 2011 02:08:39 +0000
Subject: [PATCH 44/47] drivers/net: Call netif_carrier_off at the end of the
 probe

commit 0d672e9f8ac320c6d1ea9103db6df7f99ea20361 upstream.

Without calling of netif_carrier_off at the end of the probe the operstate
is unknown when the device is initially opened. By default the carrier is
on so when the device is opened and netif_carrier_on is called the link
watch event is not fired and operstate remains zero (unknown).

This patch fixes this behavior in forcedeth and r8169.

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Acked-by: Francois Romieu <romieu@fr.zoreil.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Restricted to drivers/net/r8169.c; adjusted for omitted changes]
---
 drivers/net/r8169.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 91aa567..2c39add 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -3199,6 +3199,8 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 	}
 	pm_runtime_idle(&pdev->dev);
 
+	netif_carrier_off(dev);
+
 out:
 	return rc;
 
-- 
1.7.5.4

